bitcoin/_posts/0000-00-00-csfs-ctv-rekey-symmetry.md

```diff
@@ -0,0 +1,171 @@
+---
+comments: True
+disqusId: b2e599366ec42732af04508978f7c91203a2308d
+layout: post
+title: "CSFS Re-Keying and Laddering, Deterministic Update Rekeying, & Applications to LN-Symmetry"
+date: 2024-12-02
+---
+
+_This is a collab post with [Rearden](https://twitter.com/reardencode)._
+
+As Rearden talked about at Bitcoin++ in Austin this year, there are many ways
+to realize Lightning Symmetry using various bitcoin upgrade proposals. All of
+these methods require either an extra signing round-trip for each channel
+update, or the ability to force the hash of the settlement transaction to be
+visible with its corresponding update transaction. This can be generalized as
+the requirement that the signature authorizing a given channel be both
+rebindable (i.e. not commit to a specific prior UTXO) and commit to some
+additional data being visible for that signature to be valid.
+
+We'll start by exploring why Lightning Symmetry requires this data visibility
```

Comment:
We'll start by exploring why Lightning Symmetry requires this data visibility commitment, then dive into previously known solutions, and present a new generalized technique for using CSFS to two or more variables. Finally, we will present an optimized solution based on the principles we've developed to enable Lightning Symmetry using one extra signature, but no extra signing round-trip and without the need for concatenation or other explicit multi-commitments.

Reply:
not exactly sure, but the general N variables is of independent interest in non Lightning Symmetry contexts.